[TASK] Adjust rendering page (#347)
TYPO3-Documentation/TYPO3CMS-Guide-HowToDocument: Documentation/RenderingDocs/Index.rst

.. _render-documenation-with-docker:
.. _render-documentation-with-docker:

=========================================
Render documentation with the TYPO3 theme
=========================================

..  contents::

Rendering the :file:`Documentation` folder locally with Docker
==============================================================

> You can render the documentation of an official TYPO3 manual or a third party

Reviewer: ```suggestion
You can render the documentation of an official TYPO3 manual or a third-party
```

manual with the following steps:

#.  Clone the repository containing the documentation.

> #.  Navigate to the extension root, the directory which contains the

Reviewer: ```suggestion
#.  Navigate to the extension's root folder, the directory which contains the
```

> #.  Navigate to the directory, which contains the :file:`composer.json`

Reviewer: Why composer.json and not just project root or similar?

Author: Just to give a reference what we are talking about. If for example you are in the TYPO3 monorepository you would have to be in the sysextensions folder - just where the composer.josn ist

Author: adjusted the text

#.  Check if there is documentation to be rendered:

    A folder called :file:`Documentation` containing at least the files
    :file:`Index.rst` and :file:`guides.xml`.

> #.  Choose your prefered method of rendering (See bellow):

Reviewer: ```suggestion
#.  Choose your prefered method of rendering (See below):
```

Additionally, where is below? Maybe link this to the section?

Make sure that `Docker <https://www.docker.com/>`__ is installed on your system.

.. tabs::

    ..  group-tab:: Linux

        ..  code-block:: bash

>             mkdir -p Documentation-GENERATED-temp

Reviewer: Why one has to create the folder manually? I would expect that it is created automatically?

Author: Right now it is not created automatically, once we have fixed the container so that it works without we can remove this line. Right now there is an error if the directory is not present

            docker run --rm --pull always -v $(pwd):/project -it ghcr.io/typo3-documentation/render-guides:latest --config=Documentation
            xdg-open "Documentation-GENERATED-temp/Index.html"

    .. group-tab:: MacOS

        ..  code-block:: bash

            mkdir -p Documentation-GENERATED-temp
            docker run --rm --pull always -v $(pwd):/project -it ghcr.io/typo3-documentation/render-guides:latest --config=Documentation
            open "Documentation-GENERATED-temp/Index.html"

    ..  group-tab:: Windows

        ..  code-block:: powershell

>             start "Documentation-GENERATED-temp/Index.html"

Reviewer: What does "start"?

Author: It seems like on the mac it opens a browser and displays the file. The line was there before, I only adjusted the path. But I cannot verify it as I don't have a mac. Maybe @garvinhicking can help here

Author: Ah sorry, lost context. It is a powershell command which opens the document in a browser

Reviewer: But then it should be moved below the rendering command. But there is still the "open" command.

            docker run --rm --pull always -v ${PWD}:/project -it ghcr.io/typo3-documentation/render-guides:latest --config=Documentation
            open "Documentation-GENERATED-temp/Index.html"

> Rendering extension documentation to docs.typo3.org

Reviewer: I think "rendering" is the wrong word. Technically, on docs.typo3.org there is nothing rendered, it is the GHA. Maybe "publishing" is the better word?

===================================================

> For your documentation to be rendered to our server https://docs.typo3.org, your

Reviewer: rendered -> published ?

I would avoid "to our server". Who is meant with "our"? Is it obvious to the reader?
-> just "published on docs.typo3.org".

TYPO3 extension has to have a valid :file:`composer.json` and either a folder
called :file:`Documentation` with a :file:`Documentation/Index.rst` and
a :file:`Documentation/guides.xml` or a :file:`README.rst` / :file:`README.md` on

> the level of your :file:`composer.json`.

Reviewer: Again, why "composer.json"? It is the project/extension root?

The extension has to be publicly available on GitHub or GitLab. You have to

> introduce a :ref:`Webhook <webhook>` and the Documentation Team has to

Reviewer: "introduce" is the wrong word IMHO. Maybe "establish"?

:ref:`approve <approval-intercept>` your first rendering request on a tool

> called "Intercept".

Reviewer: Is Intercept relevant for the user? Can they do something with that knowledge? Maybe link to the docs rendering list, then it has a value.

Author: They can have a look at the public part of intercept. It is open for everyone

Author: Ok, I removed it, we probably dont have to confuse ppl here
